scripts/install.js

```diff
@@ -323,14 +317,30 @@ if (process.env.SENTRYCLI_SKIP_DOWNLOAD === '1') {
   process.exit(0);
 }
 
-(async () => {
-  try {
-    await downloadBinary();
-    await checkVersion();
-    process.exit(0);
-  } catch (e) {
-    // eslint-disable-next-line no-console
-    console.error(e.toString());
-    process.exit(1);
-  }
-})();
+const { packageName: distributionPackageName, subpath: distributionSubpath } =
+  helper.getDistributionForThisPlatform();
+
+if (distributionPackageName === undefined) {
+  helper.throwUnsupportedPlatformError();
+}
+
+try {
+  require.resolve(`${distributionPackageName}/${distributionSubpath}`);
+  // If the `resolve` call succeeds it means a binary was installed successfully via optional dependencies so we can skip the manual postinstall download.
+  process.exit(0);
+} catch (e) {
+  // Optional dependencies likely didn't get installed - proceed with fallback downloading manually
+  logger.log(
+    `Sentry CLI binary installation via optional dependencies was unsuccessful. Downloading manually instead.`
+  );
+
+  downloadBinary()
+    .then(() => checkVersion())
+    .then(() => {
+      process.exit(0);
+    })
+    .catch((e) => {
+      console.error(e);
```

Comment:
Should this be `logger.error`? I think we also need `eslint-disable-next-line no-console` in here.